frontend/src/store/cloudProfile/index.js

```diff
@@ -561,21 +564,24 @@ export const useCloudProfileStore = defineStore('cloudProfile', () => {
         severity: 'warning',
       }
     }
-    if (!version.expirationDate) {
-      return undefined
-    }
 
     const patchAvailable = kubernetesVersionIsNotLatestPatch(shootK8sVersion, shootCloudProfileName)
     const updatePathAvailable = kubernetesVersionUpdatePathAvailable(shootK8sVersion, shootCloudProfileName)
 
     let severity
-    if (!updatePathAvailable) {
-      severity = 'error'
-    } else if ((!k8sAutoPatch && patchAvailable) || !patchAvailable) {
-      severity = 'warning'
+    if (version.isExpirationWarning) {
+      if (!updatePathAvailable) {
+        severity = 'error'
+      } else if ((!k8sAutoPatch && patchAvailable) || !patchAvailable) {
+        severity = 'warning'
+      } else {
+        severity = 'info'
+      }
     } else if (k8sAutoPatch && patchAvailable) {
       severity = 'info'
-    } else {
+    }
```

Comment:
similar code is available for `expiringWorkerGroupsForShoot`. Can't we reuse the code?

Reply:
Implemented a re-use function